src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -537,6 +560,30 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
             return c;
         }
 
+#if TEST_OPERATION
+        private static void WalkOperationTree(CSharpCompilation compilation)
+        {
+            var operationWalker = TestOperationWalker.GetInstance();
+
+            foreach (var tree in compilation.SyntaxTrees)
+            {
+                var semanticModel = compilation.GetSemanticModel(tree);
+                var root = tree.GetRoot();
+
+                // TODO: check other operation root as well (property, etc.)
+                foreach (MethodDeclarationSyntax methodNode in root.DescendantNodesAndSelf().Where(n => n.Kind() == SyntaxKind.MethodDeclaration))
```

Comment:
This loop skips constructors/destructors/operators, consider using a TypeOf<BaseMethodDeclarationSyntax>() filter instead.